src/RequestExecutor/HttpRequestExecutor.ts

```diff
@@ -474,4 +468,63 @@ export class HttpRequestExecutor implements RequestExecutor {
 
     return new Request(result);
   }
+
+  private async executeRequest(
+    request: Request
+  ): Promise<Response | undefined> {
+    const { res, body } = await this.request(request);
+
+    logger.trace(
+      'received following response for request %j: headers: %j body: %s',
+      {
+        url: request.url,
+        protocol: this.protocol,
+        method: request.method
+      },
+      {
+        statusCode: res.statusCode,
+        headers: res.headers
+      },
+      body.slice(0, 500).concat(body.length > 500 ? '...' : '')
+    );
+
+    return new Response({
+      body,
+      protocol: this.protocol,
+      statusCode: res.statusCode,
+      headers: res.headers,
+      encoding: request.encoding
+    });
+  }
+
+  private tryRequestWithCertificates(
+    request: Request,
+    certs: Cert[]
+  ): Promise<Response> {
+    const requestsWithCerts: Promise<Response>[] = certs.map(
+      async (cert: Cert) => {
+        logger.debug(
+          'Executing HTTP request with following params: %j',
+          request
+        );
+        try {
+          await request.loadCert(cert);
+
+          const response = await this.executeRequest(request);
+          this.certificatesCache.add(request, cert);
+
+          return response;
+        } catch (error) {
+          const msg = Helpers.isTlsCertError(error)
+            ? `Failed to do successful request with certificate ${cert.path}. It will be excluded from list of known certificates.`
+            : `Unexpected error occured during request: ${error}`;
+          logger.warn(msg);
+          throw new Error(msg);
```

Comment:
```suggestion
          throw error;
```

Otherwise it will brake existing errors handling logic

https://github.com/NeuraLegion/bright-cli/blob/1f27ea5f7f555e70d3f561260d8d4e807f02cbb3/src/RequestExecutor/HttpRequestExecutor.ts#L122-L137

Reply:
@lsndr https://github.com/NeuraLegion/bright-cli/pull/679/commits/81d95241c02636f7ec9bda967b0bdf7d44cc04c4